packages/app-builder/src/components/Decisions/RulesDetail.tsx

```diff
@@ -188,6 +192,7 @@ function RuleFormula({
 }) {
   const astEditor = useAstBuilder({
     backendAst: formula,
+    backendValidation: evaluation,
```

Comment:
This is the core part: inject evaluation so we get errors handled the same way as validation